setup.py

```diff
@@ -16,16 +16,37 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 # ----------------------------------------------------------------------------
 
+from importlib.util import spec_from_file_location, module_from_spec
 import os
-import sys
-from metomi.isodatetime import __version__
+from pathlib import Path
 # overriding setuptools command
 # https://stackoverflow.com/a/51294311
 from setuptools.command.bdist_rpm import bdist_rpm as bdist_rpm_original
 # to parse pytest command arguments
 # https://docs.pytest.org/en/2.8.7/goodpractices.html#manual-integration
 from setuptools.command.test import test as TestCommand
 from setuptools import setup, find_namespace_packages
+import sys
+
+
+try:
+    os.chdir(os.path.dirname(__file__))
+except FileNotFoundError:
+    pass
+
+
+def get_version(module, path):
+    """Return the __version__ attr from a module sourced by FS path."""
+    spec = spec_from_file_location(module, path)
+    module = module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module.__version__
+
+
+__version__ = get_version(
+    'metomi.isodatetime',
+    Path('metomi/isodatetime/__init__.py')
```

Comment:
Looks like it doesn't work for Python 3.5:

```
Traceback (most recent call last):
  File "setup.py", line 48, in <module>
    Path('metomi/isodatetime/__init__.py')
  File "setup.py", line 40, in get_version
    spec = spec_from_file_location(module, path)
  File "<frozen importlib._bootstrap_external>", line 581, in spec_from_file_location
AttributeError: 'PosixPath' object has no attribute 'endswith'
```

Reply:
Hmmm? It works for me on Python3.4 (don't have 3.5 at hand):

```console
$ python3.4
Python 3.4.10 (default, Apr  8 2019, 02:17:36) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-36)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> Path('metomi/isodatetime/__init__.py')
PosixPath('metomi/isodatetime/__init__.py')
```

Reply:
Ahhh, got it, the `spec_from_file_location` function only picked up Pathlib support at 3.6!

https://docs.python.org/3/library/importlib.html#importlib.util.spec_from_file_location

Reply:
Have added Python3.5 support as it was simple to do.